crates/sui-core/src/traffic_controller/nodefw_client.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021, Facebook, Inc. and its affiliates
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddresses {
+    pub addresses: Vec<BlockAddress>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct BlockAddress {
+    pub source_address: String,
+    pub destination_port: u16,
+    pub ttl: u64,
+}
+
+pub struct NodeFWClient {
+    client: reqwest::Client,
+    remote_fw_url: String,
+}
+
+impl NodeFWClient {
+    pub fn new(remote_fw_url: String) -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            remote_fw_url,
+        }
+    }
+
+    pub async fn block_addresses(&self, addresses: BlockAddresses) -> Result<(), reqwest::Error> {
+        self.client
+            .post(&format!("{}/block_addresses", self.remote_fw_url))
+            .json(&addresses)
+            .send()
+            .await?;
```

Comment:
i would check that it is a StatusCode::ACCEPTED

https://docs.rs/reqwest/latest/reqwest/struct.StatusCode.html#associatedconstant.ACCEPTED

Reply:
Moved to using `Response::error_for_status()` which will convert any client or server http error code into an `Err`. Imo, we shouldn't care about the specific success code returned by the server here (e.g. StatusCode::ACCEPTED` vs `StatusCode::OK`) as long as it's success